engine-comparison/report-gen/generate-report.go

```diff
@@ -66,56 +63,63 @@ func handleTrialCSV(this_reader *csv.Reader, records [][]string, fengine os.File
 	return records
 }
 
+func handleFengine(fengine os.FileInfo, current_path string, desired_report_fname string) {
+	this_fe_file, err := os.Create(path.Join(current_path, fengine.Name(), desired_report_fname))
+	checkErr(err)
+	defer this_fe_file.Close()
+	this_fe_writer := csv.NewWriter(this_fe_file)
+
+	// Create matrix, to eventually become a CSV
+	records := [][]string{{"time"}}
+
+	// Enter sub-directories
+	potential_trials, err := ioutil.ReadDir(path.Join(current_path, fengine.Name()))
+	checkErr(err)
+	trials := onlyDirectories(potential_trials)
+
+	num_record_columns := len(trials) + 1
+	for j, trial := range trials {
+		// Create fds
+		this_file, err := os.Open(path.Join(current_path, fengine.Name(), trial.Name(), desired_report_fname))
+		checkErr(err)
+		defer this_file.Close()
```

Comment:
Depending on the number of trials here, we could be opening a lot of files that remain open until handleFengine() returns.

Reply:
Certainly true; `defer` seems to be strictly less efficient, and is primarily useful for readability I suppose. Definitely worse in a loop